### ISSUE TITLE
E0001 Error transforming template: Provided SourceAccessConfigurations cannot be parsed into a list.

### ISSUE THREAD

**Issue by woowig** [aws-cloudformation/cfn-lint#2395]
### CloudFormation Lint Version

0.65.1

### What operating system are you using?

Mac

### Describe the bug

I want to create a rabbitmq trigger lambda function with AWS SAM.
I wrote a list in SourceAccessConfigurations like this and I succeeded to deploy with cloudformation.

```yaml
      Events:
        HelloWorld:
          Type: MQ
          Properties:
            Broker: !GetAtt HelloWorldBroker.Arn
            Queues:
              - HelloWorld
            SourceAccessConfigurations:
              - Type: BASIC_AUTH
                URI: !Ref HelloWorldSecret
            Enabled: true
```

But cfn-lint reported it's invalid.
The error message below is shown.

```
E0001 Error transforming template: Resource with id [HelloWorldFunction] is invalid. Event with id [HelloWorld] is invalid. Provided SourceAccessConfigurations cannot be parsed into a list.
template.yaml:1:1
```

### Expected behavior

the error message shall not be shown.

### Reproduction template

```yaml
AWSTemplateFormatVersion: '2010-09-09'
Transform: AWS::Serverless-2016-10-31
Description: MQ + Lambda

Resources:
  HelloWorldSecret:
    Type: AWS::SecretsManager::Secret
    Properties:
      GenerateSecretString:
        GenerateStringKey: password
        PasswordLength: 32
        SecretStringTemplate: '{"username": "admin"}'

  HelloWorldBroker:
    Type: AWS::AmazonMQ::Broker
    Properties:
      AutoMinorVersionUpgrade: false
      BrokerName: !Sub "${AWS::StackName}-HelloWorldBroker"
      DeploymentMode: SINGLE_INSTANCE
      EngineType: RabbitMQ
      EngineVersion: '3.9.16'
      HostInstanceType: mq.t3.micro
      PubliclyAccessible: true
      Users:
        - Password: !Sub '{{resolve:secretsmanager:${HelloWorldSecret}:SecretString:password}}'
          Username: !Sub '{{resolve:secretsmanager:${HelloWorldSecret}:SecretString:username}}'

  HelloWorldFunction:
    Type: AWS::Serverless::Function
    Properties:
      Handler: index.handler
      Runtime: nodejs16.x
      Architectures:
        - x86_64
      InlineCode: >
        exports.handler = async (event, context) => {
          console.log(JSON.stringify(event))
        }
      Events:
        HelloWorld:
          Type: MQ
          Properties:
            Broker: !GetAtt HelloWorldBroker.Arn
            Queues:
              - HelloWorld
            SourceAccessConfigurations:
              - Type: BASIC_AUTH
                URI: !Ref HelloWorldSecret
            Enabled: true

Outputs:
  HelloWorldEndpoint:
    Value: !Select [0, !GetAtt HelloWorldBroker.AmqpEndpoints]
  HelloWorldSecret:
    Value: !Ref HelloWorldSecret
```

**Comment by PatMyron** [comment]
reproduced with last few `cfn-lint`/`aws-sam-translator` versions:
https://github.com/aws/serverless-application-model/blob/a25933379e1cad3d0df4b35729ee2ec335402fdf/samtranslator/model/eventsources/pull.py#L279

**Comment by kddejong** [comment]
SAM translator has been released and is updated with recent versions of cfn-lint
